### PR TITLE
Remove pcpn_cum_amt from WAV matviews

### LINKED ISSUE #215: Support alternate monthly mean calculations for tipping bucket stations
Stations with the `network_variable_name` of `cum_pcpn_amt` seem to measure precipitation by accumulation - that is, they report how much total precipitation has fallen since the last time the instrument reset on an hourly basis, instead of reporting how much precipitation has fallen that hour. Here are a couple representative graphs:

tip
tip2
tip3

Physically it seems likely these are "tippng bucket" stations, measuring rainfall via a bucket that accumulates water until it is full, and tips out to reset.

In order to calculate monthly precipitation observed at these stations, we do not want to average hourly or daily reported values; this leads to months where, in the case of the second graph, every day or hour sees 1600 mm of precipitation! I believe the right calculation goes something like:

```
month_max = maximum value recorded during month
month_min = smallest value recorded during month
month_last = last value recorded during month
month_first = first value recorded during month

if month_max > month_last: #bucket "tipped" this month
   return (month_last - month_min) + (month_max - month_first)
else: bucket did not "tip" this month
    return month_last - month_first
```

In order for the back end to calculate this special case reasonably quickly, we need a new materialized view with `month_max`, `month_min`, `month_last`, and `month_first` for each station and month using the `cum_pcpn_amt` variable, as well as the earliest and latest days each month for which data is available so we can prorate it if there is less than a full month's data in the database.

### LINKED ISSUE #216: Recreate monthly_total_precipitation_mv as a native matview
We'd like to change the construction of `monthly_total_precipitation_mv` (currently a table) to exclude `cum_pcpn_amt` and include instead `pcpn_amt_pst24hrs` as using cumulative precipitation totals to construct monthly statistics is throwing things off. 

The first step will be to import `monthly_total_precipitation_mv` as a native matview managed by the ORM, then we can make a couple changes to variable inclusion.

### PR DESCRIPTION
**Pull request by corviday**
The main goal of this PR is to remove the `pcpn_amt_cum` variable, which is reported by some stations with low-accuracy rugged tipping bucket sensors, from the `monthly_total_precipitation` matview, which is used by the Weather Anomaly Viewer. After discussion with sensor network personnel, we have realized these tipping bucket sensors are not accurate enough for this particular usage.

The ancillary goal of this PR, on which most of the linecount was spent, was to update the remaining "manual" matviews (tables maintained by a script) to "native" matviews that run on the newer postgres version. Most of the query verbiage here is just copied from one file to another, with the minimal changes needed to make it work as a native matview. 

[Demo](https://services.pacificclimate.org/dev/weather-anomaly-viewer/) - running against a version of the database from January, so you'll have to click back to January to see any data.

resolve #216 
resolve #215 (though we ended up removing the tipping bucket calculations entirely)

### CODE REVIEW

**Comment by Nospamas** [comment]
I've bumped the [beehive](http://beehive.pacificclimate.org/weather-anomaly-viewer/#) version of dev to the latest via [commit](https://github.com/pacificclimate/portainer-deployment/commit/191392efc0182b65d452ac1e3582c7ae1fa1bd9a)

**Comment by Nospamas** [comment]
[Updated the db as well](https://github.com/pacificclimate/portainer-deployment/commit/2d011b8eb96ec3e1fa2c2ff14252a05db1a9e35b) :)

File: pycds/orm/manual_matviews/__init__.py [hunk 1]
```diff
@@ -19,21 +19,9 @@
 export the latest version. When a PyCDS release is created, it will "freeze"
 this set of views. Following any PyCDS release, further migrations and further
 releases will "freeze" later sets of views.
+
+At present all manual matviews have been updated to native matviews, and this
+file is residual.
```
Comment (rod-glover) [inline pycds/orm/manual_matviews/__init__.py:24]: Is there a reason we can't delete the file altogether?

File: tests/alembic_migrations/versions/v_081f17262852_convert_monthly_total_precipitation_mv_/test_smoke.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+"""Smoke tests:
+- Upgrade adds columns to matview
+- Downgrade drops columns from matview
+"""
```
Comment (rod-glover) [inline tests/alembic_migrations/versions/v_081f17262852_convert_monthly_total_precipitation_mv_/test_smoke.py:4]: This docstring is kind of generically true of many smoke tests, but I think it is not relevant to this particular test. It looks to me as if what is tested is the existence of the matview and its associated indexes.

**Review by rod-glover — COMMENTED** [review]
I cross-checked against the old matviews deployed to production and the Jan 2024 stations mostly show the same anomalies, but not all (?!?). 

Most seem to be like this, where anomalies are identical. (Demo on left; prod on right.)

image

But some disagree:

image

image

Perhaps there is a good reason? Like maybe not enough records in the test database? It would be good to be able to know what is going on here. Would it be feasible to test the results of old and new against each other on the same database?

**Comment by rod-glover** [comment]
Hmm, I realized a little belatedly that this PR is confined to precip. And these disagreements are in the temperature data. That may or may not signal a problem ... somewhere.

**Comment by corviday** [comment]
Thank you for comparing actual data!

The issue might just be records present in the live database but not the test database (which is a copy of the live database at whatever arbitrary point I asked Tom to make me one).

It's also possible I made some kind of error setting up the new matviews, but I would be a little surprised because most of the create-the-table was just wholesale copied, but maybe there are subtleties of matview versus table I didn't correct.

**Comment by corviday** [comment]
Next steps:

* determine whether temperature errors observed by @rod-glover  arise from the new pyCDS code or incompleteness of the test database by downgrading the test database and the dev instance and seeing if the temperature errors are still present
* remove manage_views script and the manial_matviews __init__.py
* write plan for migration and get @rod-glover 's feedback

File: pycds/orm/manual_matviews/__init__.py (hunk 1 above)
Comment (corviday) [reply pycds/orm/manual_matviews/__init__.py:24]: I thought it might be needed to support upgrading and downgrading?

**Comment by corviday** [comment]
I have migrated the database back to the previous version and looked up the Mica and 50 mile stations, and their value still diverge from the production viewer, so I think it is safe to say these errors were caused by observation 

records missing from the test database, and not the migration itself.

**Comment by corviday** [comment]
With extensive help from @rod-glover , I have verified that crmprtd can add data to a database migrated to this version of pyCDS (even without upgrade crmprtd).

**Pull request merged**